.pre-commit-config.yaml

```diff
@@ -64,23 +64,8 @@ repos:
         - APScheduler~=3.10.1
         - cachetools~=5.3.0
         - . # this basically does `pip install -e .`
--   repo: https://github.com/asottile/pyupgrade
-    rev: v3.3.1
-    hooks:
-    -   id: pyupgrade
-        files: ^(telegram|examples|tests|docs)/.*\.py$
-        args:
-          - --py37-plus
--   repo: https://github.com/pycqa/isort
-    rev: 5.12.0
-    hooks:
-    -   id: isort
-        name: isort
-        args:
-        - --diff
-        - --check
```

Comment:
What is the compatibility status here? IIRC @lemontree210 reported that isort detects some things that ruff doesn't. moreover, ruff is still far away from being stable. I'm not quite comfortable with dropping pyupgrade and isort completely yet …

Reply:
I just said that isort was more informative: it shows exactly what the problem is, whereas ruff just states that there is a sorting problem in a given file.

I decided to keep isort in my own project too.

Reply:
ruff's readme does say it replace pyupgrade and isort.. And also I've never had or seen pyupgrade fail once btw

Reply:
Update: https://twitter.com/charliermarsh/status/1637512274716893187?s=20 (and the next tweet)

Based on the issue linked in the FAQ for isort (https://github.com/charliermarsh/ruff/issues/1381, https://github.com/charliermarsh/ruff/issues/2104), there are slight differences how ruff handles it, which I'm okay with.

The pyupgrade one I believe we can definitely drop, based on the tweet.

Reply:
Thanks for the links! Then from compatibility POV I'd be okay.

from stability pov I'm still cautious though. I just had a look at the contributiors page https://github.com/charliermarsh/ruff/graphs/contributors and that makes me more reserved for two reasons:

* the project is barely half a year old
* there is one main contributor

meaning that if the main contributor can no longer advance the development for whichever reason, the project is basically stuck at a way-before-stable status … I'd very much like the project to get more mature before relying too much on it.

Finally, in my experience isort & pyupgrade usually don't take much time - that's mostly pylint.